core/src/test/java/org/apache/calcite/test/RelOptTestBase.java

```diff
@@ -236,6 +238,7 @@ protected void checkPlanning(Tester tester, HepProgram preProgram,
       }
     }
     SqlToRelTestBase.assertValid(r);
+    return r;
```

Comment:
You should not return `relNode` from this method, you can reference this code to get a `relNode` with rule fired [1]

[1] https://github.com/apache/calcite/blob/df3dc81f23166c15e72d5414944c69a35c2bd110/core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java#L311